### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by MasonProtter**
I know there's not anything particularly new here, but I thought it'd be good to show an example where you construct an `OffsetArray` from a preexisting array instead of an `undef`. This might be redundant for experienced users but I think it's helpful for newcomers. The relativistic notation was just the first example that came to mind where `0` based indexing was natural. 

It can be changed to something more natural or accessible if needed.

### CODE REVIEW

**Comment by timholy** [comment]
Hmm, yes, the current README is indeed pretty anemic. :+1: 

Why do you insert `:t` in the constructor instead of prior? (Just thinking we might as well make this pretty simple.) Also, since I imagine most people will be interested in numeric arrays, what about something like

```julia
η = 0.1
v1 = [sinh(η), cosh(η), 0, 0]
v0 = OffsetArray(v1, 0:3)
```

just to separate the values of the array from setting the indexing?

I don't feel attached to these suggestions, just wondering. And your version might have some advantages, so I'd be happy to merge this as-is.

**Comment by MasonProtter** [comment]
I did `[:t, [:x, :y, :z]...]` because I thought might highlight that I’m interested in a spatial, one-based vector `r = [:x, :y, :z]` and a zero-based spacetime vector `x`. I like this because it emphasizes that 0-based and 1-based arrays might be useful for two related quantities in *one* problem. 

Perhaps it’d e good to have the symbolic example as well as the numeric example? Or I can just switch to your numeric example.

**Comment by MasonProtter** [comment]
I've added an exmple where one may want to express the coefficients of the Laurent polynomial 
```
6/x + 5 - 2 * x + 3 * x^2 + x^3
```
as a `-1`-based array, since if one counts from `-1`, the `nth` coefficient corresponds to the `nth` power of `x`. This has the advantage of introducing the `eachindex` function which is very useful when working with any AbstractArray.

File: README.md [hunk 1]
```diff
@@ -20,6 +20,56 @@ julia> y[-1,-7,-128,-5,-1,-3,-2,-1] += 5
...
+ :z
+```
+
+## Example: Polynomials
+Suppose one wants to represent the Laurent polynomial
+```
+6/x + 5 - 2*x + 3*x^2 + x^3
+```
+in julia. The coefficients of this polynomial are a naturally `-1` based list, since the `n`th element of the list 
+(counting from `-1`) `6, 5, -2, 3, 1` is the coefficient corresponding to the `n`th of `x`. This Laurent polynomial can be evaluated at say `x = 2` as follows.
```
Comment (timholy) [inline README.md]: `n`th *power* of `x`

**Comment by timholy** [comment]
Test failures addressed in #62

File: README.md (hunk 1 above)
Comment (MasonProtter) [reply README.md]: Good catch! Fixed.

**Comment by timholy** [comment]
Thanks for an excellent and much-needed enhancement!

**Pull request merged**